.github/workflows/ci.yml

```diff
@@ -25,34 +26,29 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
-      - uses: haskell/actions/setup@v1
+      - if: ${{ runner.os == 'Linux' }}
+        run: |
+          sudo apt-get update
+          sudo apt-get install zsh
+      - if: ${{ runner.os == 'macOS' }}
+        run: |
+          brew update
+          brew install zsh
+      - uses: haskell/actions/setup@v2
         with:
           ghc-version: ${{ matrix.ghc }}
           cabal-version: "3.6"
-      - run: |
-          if [ "$RUNNER_OS" == "Linux" ]
-          then
-            sudo apt-get update
-            sudo apt-get install zsh
-          else
-            brew update
-            brew install zsh
-          fi
-      - run: cabal update
+        id: setup-haskell
       - run: cabal configure --enable-tests
       - run: cabal freeze
       - uses: actions/cache@v3
         with:
-          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('*.cabal', 'cabal.project.freeze') }}
-          restore-keys: ${{ runner.os }}-${{ matrix.ghc }}-
+          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
           path: |
-            ~/.cabal/packages
-            ~/.cabal/store
+            ${{ steps.setup-haskell.outputs.cabal-store }}
             dist-newstyle
-      - run: cabal build --only-dependencies
+          restore-keys: ${{ runner.os }}-${{ matrix.ghc }}-
       - run: cabal build
       - run: cabal test
-      - run: |
-          # Fixes issue with different haddock coverage with different ghc versions https://github.com/haskell/haddock/issues/123
-          cabal haddock | grep "100%" | wc -l | grep -E "[4-9]|[1-9][0-9]+" ||
-          (echo "Haddock failed with exit code 1. Have you checked that the minimum of 4 modules with 100% documentation is fulfilled?" && false)
+      - run: cabal haddock
+      - run: cabal sdist
```

Comment:
@sestrella Why do we need this step?

Reply:
@CristhianMotoche it is not strictly necessary right now, but the output produced by this command could be upload to Hackage

> cabal sdist [FLAGS] [TARGETS] takes the crucial files needed to build TARGETS and puts them into an archive format ready for upload to Hackage. These archives are stable and two archives of the same format built from the same source will hash to the same value.

https://cabal.readthedocs.io/en/latest/cabal-commands.html?highlight=sdist#cabal-sdist